scanners/generic/generic.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import os
+import pprint
+import shutil
+
+from scanners import RapidastScanner
+from scanners import State
+
+
+CLASSNAME = "Generic"
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+# Helper: absolute path to this directory (which is not the current directory)
+# Useful for finding files in this directory
+MODULE_DIR = os.path.dirname(__file__)
+
+
+class Generic(RapidastScanner):
+    ###############################################################
+    # PRIVATE CONSTANTS                                           #
+    # Accessed by Generic only                                    #
+    ###############################################################
+
+    ###############################################################
+    # PROTECTED CONSTANTS                                         #
+    # Accessed by parent Generic object                           #
+    ###############################################################
+
+    def __init__(self, config, ident):
+        logging.debug("Initializing Generic scanner")
+        super().__init__(config, ident)
+
+        # create a temporary directory (cleaned up during cleanup)
+        self.workdir = self._create_temp_dir("workdir")
+
+        # The command to run (excluding the "container" layer)
+        self.generic_cli = []
+
+    ###############################################################
+    # PUBLIC METHODS                                              #
+    # Called via inheritence only                                 #
+    ###############################################################
+
+    def setup(self):
+        """Prepares everything:
+        - the command line to run
+        - environment variables
+        - files & directory
+
+        This code handles only the "Generic" layer, independently of the container used.
+        This method should not be called directly, but only via super() from a child's setup()
+        """
+        logging.info("Preparing Generic configuration")
+
+    def run(self):
+        """This code handles only the "Generic" layer, independently of the container used.
+        This method should not be called directly, but only via super() from a child's setup()
+        This method is currently empty as running entirely depends on the containment
+        """
+        pass
+
+    def postprocess(self):
+        logging.info(f"Extracting report, storing in {self.results_dir}")
+        result = self.my_conf("container.results")
+        try:
+            os.makedirs(self.results_dir)
+            if os.path.isdir(result):
+                shutil.copytree(result, self.results_dir)
```

Comment:
, dirs_exist_ok=True is necessary. otherwise, an error occurs like: 
`ERROR:Unable to save results: [Errno 17] File exists: './results/app4trivy/DAST-20230828-233412-RapiDAST-app4trivy/generic'`

Reply:
updated